[Doc] Added Accelerate Python (#5940)
taichi-dev/taichi: docs/lang/articles/get-started/accelerate_python.md

   ```python
   @ti.func
   def is_prime(n: int):
       result = True
       for k in range(2, int(n ** 0.5) + 1):
           if n % k == 0:
               result = False
               break
       return result

   @ti.kernel
   def count_primes(n: int) -> int:
       count = 0
       for k in range(2, n):
           if is_prime(k):
               count += 1

       return count
   ```
4. Rerun **count_primes.py**：
   ```bash
   time python count_primes.py
   ```
   *The calculation speed is six times up (2.235/0.363).*
   ```bash
   78498

   real        0m0.363s
   user        0m0.546s
   sys        0m0.179s
   ```
5.  Increase `N` tenfold to `10,000,000` and rerun **count_primes.py**:
   *The calculation time with Taichi is 0.8s vs. 55s with Python only. The calculation speed with Taichi is 70x up.*

6. Change Taichi's backend from CPU to GPU and give it a rerun:
   ```python
   ti.init(arch=ti.gpu)
   ```
   *The calculation time with Taichi is 0.45s vs. 55s with Python only. The calculation speed with Taichi is taken further to 120x up.*

## Dynamic programming: longest common subsequence

> Dynamic programming (DP) is a well-known algorithm. The core philosophy behind it is to sacrifice some storage space for less execution time and store intermediate results to avoid repetitive computation. In the following section, we will walk you through a complete implementation of DP, and demonstrate another area where Taichi can make a real 'acceleration'.

Reviewer: DP is not *an* algorithm, it's a  philosophy of solving problems ...

---
sidebar_position: 2
---

# Accelerate Python with Taichi

Taichi is a domain-specific language *embedded* in Python. One of its key features is that Taichi can accelerate computation-intensive Python programs and help these programs [achieve comparable performance to C/C++ or even CUDA](https://docs.taichi-lang.org/blog/is-taichi-lang-comparable-to-or-even-faster-than-cuda). This makes Taichi much better positioned in the area of scientific computation.

> In the following sections, we provide two examples to give you a sense as to how much acceleration Taichi can bring to your Python programs.

Reviewer: ```suggestion
In the following sections, we provide two examples to give you a sense of how much performance gain Taichi can bring to your Python programs.
```

Author: "as to" is grammatically correct.

## Count the primes

> Large-scale or nested for loops in Python always leads to poor runtime performance. The following code is a simple demo of counting the primes within a specified range and involves nested for loops (see [here](https://github.com/taichi-dev/faster-python-with-taichi/blob/main/count_primes.py) for the complete version). Simply by importing Taichi or switching to Taichi's GPU backends, you will see a significant boost to the overall performance:

Reviewer: ```suggestion
Large-scale or nested for loops in Python always leads to poor runtime performance. The following simple demo counts the primes within a specified range and involves nested for loops (see [here](https://github.com/taichi-dev/faster-python-with-taichi/blob/main/count_primes.py) for the complete version). Simply by importing Taichi or switching to Taichi's GPU backends, you will see a significant boost to the overall performance:
```

   *Now you get the following program:*

```python
import taichi as ti
import numpy as np

ti.init(arch=ti.cpu)

benchmark = True

N = 15000

f = ti.field(dtype=ti.i32, shape=(N + 1, N + 1))

if benchmark:
    a_numpy = np.random.randint(0, 100, N, dtype=np.int32)
    b_numpy = np.random.randint(0, 100, N, dtype=np.int32)
else:
    a_numpy = np.array([0, 1, 0, 2, 4, 3, 1, 2, 1], dtype=np.int32)
    b_numpy = np.array([4, 0, 1, 4, 5, 3, 1, 2], dtype=np.int32)

@ti.kernel
def compute_lcs(a: ti.types.ndarray(), b: ti.types.ndarray()) -> ti.i32:
    len_a, len_b = a.shape[0], b.shape[0]

    ti.loop_config(serialize=True) # Disable auto-parallelism in Taichi
    for i in range(1, len_a + 1):
        for j in range(1, len_b + 1):
               f[i, j] = max(f[i - 1, j - 1] + (a[i - 1] == b[j - 1]),
                          max(f[i - 1, j], f[i, j - 1]))

    return f[len_a, len_b]

print(compute_lcs(a_numpy, b_numpy))
```

8. Save the above code as **lcs.py** and run:

```bash
time python lcs.py
```
   *The system prints the length of the LCS, along with the execution time.*

```bash
2721

real        0m1.409s
user        0m1.112s
sys        0m0.549s
```

> In [this repo](https://github.com/taichi-dev/faster-python-with-taichi/blob/main/lcs.py), we provide our implementation of this dynamic planning algorithm in Taichi and NumPy:

Reviewer: ```suggestion
In [this repo](https://github.com/taichi-dev/faster-python-with-taichi/blob/main/lcs.py), we provide our implementation of this dynamic planning algorithm in Taichi and NumPy: 
(the actual time cost may vary on your machine, but we believe the performance enhancement is comparable with ours)
```
